src/View/Helper/BreadcrumbsHelper.php

```diff
@@ -0,0 +1,268 @@
+     * The crumbs list.
+     *
+     * @var array
+     */
+    protected $crumbs = [];
+
+    /**
+     * Add a crumb to the trail.
+     *
+     * @param string $title Title of the crumb
+     * @param string|array|null $link Link of the crumb. Either a string, an array of route params to pass to
+     * Url::build() or null / empty if the crumb does not have a link
+     * @param array $options Array of options
+     * @return $this
+     */
+    public function add($title, $link = null, array $options = [])
+    {
+        $this->crumbs[] = ['title' => $title, 'link' => $link, 'options' => $options];
+
+        return $this;
+    }
+
+    /**
+     * Prepend a crumb to the start of the queue.
+     *
+     * @param string $title Title of the crumb
+     * @param string|array|null $link Link of the crumb. Either a string, an array of route params to pass to
+     * Url::build() or null / empty if the crumb does not have a link
+     * @param array $options Array of options
+     * @return $this
+     */
+    public function prepend($title, $link = null, array $options = [])
+    {
+        array_unshift($this->crumbs, ['title' => $title, 'link' => $link, 'options' => $options]);
+
+        return $this;
+    }
+
+    /**
+     * Insert a crumb at a specific index.
+     *
+     * If the index already exists, the new crumb will be inserted,
+     * and the existing element will be shifted one index greater.
+     *
+     * @param int $index The index to insert at.
+     * @param string $title Title of the crumb
+     * @param string|array|null $link Link of the crumb. Either a string, an array of route params to pass to
+     * Url::build() or null / empty if the crumb does not have a link
+     * @param array $options Array of options
+     * @return $this
+     */
+    public function insertAt($index, $title, $link = null, array $options = [])
+    {
+        array_splice($this->crumbs, $index, 0, [['title' => $title, 'link' => $link, 'options' => $options]]);
+
+        return $this;
+    }
+
+    /**
+     * Insert a crumb before the first matching crumb with the specified title.
+     *
+     * Finds the index of the first middleware that matches the provided class,
+     * and inserts the supplied callable before it.
+     *
+     * @param string $matchingTitle The title of the crumb you want to insert this one before
+     * @param string $title Title of the crumb
+     * @param string|array|null $link Link of the crumb. Either a string, an array of route params to pass to
+     * Url::build() or null / empty if the crumb does not have a link
+     * @param array $options Array of options
+     * @return $this
+     */
+    public function insertBefore($matchingTitle, $title, $link = null, array $options = [])
+    {
+        $found = false;
+        foreach ($this->crumbs as $key => $crumb) {
+            if ($crumb['title'] === $matchingTitle) {
+                $found = true;
+                break;
+            }
+        }
```

Comment:
you could extract the duplicate code in both insertBefore and insertAfter into a protected method.
This way it would also be easier to break early by just returning true.